Code/tests/test_autopkglib.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+
+import imp
+import plistlib
+import unittest
+from textwrap import dedent
+
+from mock import patch
+
+
+# DO NOT MOVE THIS! This needs to happen BEFORE importing autopkglib
+patch("autopkglib.memoize", lambda x: x).start()
```

Comment:
It took me a while to figure this out. Memoizing wraps the function at module import time, so the values for `is_mac()`, `is_windows()`, etc. were all predefined at import time. To do effective unit testing on a memoized function, we have to essentially apply a patch on memoize itself. By doing this here, we are overlaying a patch for `autopkglib.memoize` that always just returns what it's called with, rather than using the one defined in autopkglib.

This has to be defined prior to autopkglib being imported, so the patch is present on first load.